vault.js

```diff
@@ -1,4 +1,30 @@
+/*jshint esversion: 6 */
 'use strict';
-module.exports = function() {
-  
-};
\ No newline at end of file
+function VaultGenerator() {
+  let myVault = {};
+  return {
+    setValue : setValue,
+    getValue : getValue
+  };
+
+  function setValue ( key, value ) {
+    myVault[key] = value;
+  }
+
+  function getValue ( key ) {
```

Comment:
This implementation can be simplified.
There is an edge case that would trip this solution.
```
const vault = VaultGenerator();

vault.getValue('toString'); // [Function toString]
// should return null

```